bin/circle/search-circle-builds

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env ruby
+# A script to analyze a range of CircleCI builds for known failures.
+# For available options, run ./search-circle-builds -h
+
+require 'pp'
+require 'json'
+require 'memoist'
+require 'open-uri'
+require 'optparse'
+
+CIRCLE_API_BASE = 'https://circleci.com/api/v1.1/project/github/code-dot-org/code-dot-org'
+
+KNOWN_FAIL_TYPES = [
+    {
+        search_string: 'ConnectionTimeoutError',
+        step_string: 'run_tests',
+        container: 1
+    },
+    {
+        search_string: 'io_fillbuf',
+        step_string: 'run_tests',
+        container: 1
+    },
+]
```

Comment:
Hey Brian, after chasing a bunch of these issues around yesterday, I see why you're doing this now. It would still be nice to be able to specify on the command line what to grep for, but it does seem valuable to add this as is.

Would you be willing to add a clause to detect the "invalid base64" problem to the known fail types? After that, LGTM! Happy to take another look if you decide to add custom grepping.